src/pool/srv_pool.c

```diff
@@ -6653,6 +6691,31 @@ pool_svc_update_map_internal(struct pool_svc *svc, unsigned int opc,
 		goto out_map;
 	}
 
+	/* For SWIM exclude, don't change pool map if the pw_rf is broken or is going to be broken,
+	 * with CRIT log message to ask administrator to bring back the engine.
+	 */
+	if (src == MUS_SWIM && opc == MAP_EXCLUDE) {
+		d_rank_t	rank;
+		int		failed_cnt;
+
+		rc = pool_map_update_failed_cnt(map);
+		if (rc) {
+			DL_ERROR(rc, "pool_map_update_failed_cnt failed.");
+			goto out_map;
+		}
+
+		D_ASSERT(tgt_addrs->pta_number == 1);
+		rank = tgt_addrs->pta_addrs->pta_rank;
+		failed_cnt = pool_map_get_failed_cnt(map, PO_COMP_TP_NODE);
+		D_INFO("SWIM exclude rank %d, failed_cnt %d\n", rank, failed_cnt);
+		if (failed_cnt > pw_rf) {
```

Comment:
So `pw_rf` is based on `rank` or `node` ?

Reply:
on NODE.